Detectors/MUON/MCH/Simulation/src/Stepper.cxx

```diff
@@ -0,0 +1,105 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#include "Stepper.h"
+
+#include "SimulationDataFormat/Stack.h"
+#include "SimulationDataFormat/TrackReference.h"
+#include "TGeoManager.h"
+#include "TVirtualMC.h"
+#include "TVirtualMCStack.h"
+#include "TMCProcess.h"
+#include <iomanip>
+#include "TArrayI.h"
+
+#include "FairRootManager.h"
+
+namespace o2
+{
+namespace mch
+{
+
+Stepper::Stepper() : mHits{ new std::vector<o2::mch::Hit>(20) } {}
+Stepper::~Stepper()
+{
+  delete mHits;
+  showProcesses();
+}
+
+void Stepper::showProcesses() {
+  std::cout << "--- MCH Stepper : processes seen\n";
+  for (int i = 0; i < kMaxMCProcess; i++) {
+    if (mProcessCodes[i] > 0) {
+      std::cout << "---" << std::setw(40) << TMCProcessName[i] << " " << std::setprecision(2) << 100.0*mProcessCodes[i]/mNofSteps << "% \n";
+    }
+  }
+}
+
+void Stepper::countProcesses(const TVirtualMC& vmc)
```

Comment:
these kind of things are already monitored by MCStepLogger. Is there a separate need to monitor this here?

Reply:
Well, the info from the StepLogger can be analysed in many ways. We are working on a convenient tool to extract any high-level plot. Please also just communicate your needs for the StepLogger.

With putting the analysis here, you are either always accumulating this information (even if it is not needed) or there is at least an additional runtime check. So unless this info influences your hit creation, I would not put it here.

Reply:
( For me this is not a blocker to merge this PR ... but I would really like to avoid adding such debug code to the final processing ).

Reply:
I understand and agree : I acknowledge it's a lot easier not to introduce some code in the first place than it is to remove it afterwards ;-). Will remove that part.

What I'd need from the StepLogger is a (easy) way to get a summary, once (at the end of the process only), of the different type of processes that took place in the MCH sensitive volumes.

Reply:
You will get that: Benedikt is working on an analysis tool that allows to put easy customizable analysis of the step-logger data.